argon2/src/lib.rs

```diff
@@ -337,126 +342,133 @@ impl<'key> Argon2<'key> {
                         && pass == 0
                         && slice < SYNC_POINTS / 2);
 
-                for lane in 0..lanes {
-                    let mut address_block = Block::default();
-                    let mut input_block = Block::default();
-                    let zero_block = Block::default();
-
-                    if data_independent_addressing {
-                        input_block.as_mut()[..6].copy_from_slice(&[
-                            pass as u64,
-                            lane as u64,
-                            slice as u64,
-                            memory_blocks.len() as u64,
-                            iterations as u64,
-                            self.algorithm as u64,
-                        ]);
-                    }
-
-                    let first_block = if pass == 0 && slice == 0 {
+                memory_blocks
+                    .segment_views(slice, lanes)
+                    .for_each(|mut memory_view| {
+                        let lane = memory_view.lane();
```

Comment:
Please note that this `fill_blocks` diff is very noisy, due to a necessary indentation change and then rustfmt.

The only "real" changes in this function are the use of the segment view iterator (here), the accessing of memory through the segment view API instead of through indexing of the `memory_blocks` slice (bellow), and changing `memory_blocks` to be mutable (above).